components/MapDisplay.js

```diff
@@ -0,0 +1,69 @@
+class MapDisplay {
+  constructor(app) {
+    this.app = app;
+    this.latitudeElement = document.querySelector(".latitude-text");
+    this.longitudeElement = document.querySelector(".longitude-text");
+    this.loadingText = document.querySelector(".loading-text");
+
+    this.map = null;
+    this.placemark = null;
+  }
+
+  successCallback(position) {
+    const latitude = position.coords.latitude;
+    const longitude = position.coords.longitude;
+
+    this.latitudeElement.textContent = `Latitude: ${latitude
+      .toString()
+      .slice(0, 5)
+      .replace(".", "° ")}'`;
+    this.longitudeElement.textContent = `Longitude: ${longitude
+      .toString()
+      .slice(0, 5)
+      .replace(".", "° ")}'`;
+
+    ymaps.ready(() => {
+      if (!this.map) {
+        this.map = new ymaps.Map("map", {
+          center: [latitude, longitude],
+          zoom: 10,
+        });
+
+        this.placemark = new ymaps.Placemark([latitude, longitude], {
+          hintContent: "Ваше текущее местоположение",
+        });
+        this.map.geoObjects.add(this.placemark);
+      } else {
+        this.map.setCenter([latitude, longitude]);
+        this.placemark.geometry.setCoordinates([latitude, longitude]);
+      }
+
+      this.loadingText.style.display = "none";
+    });
+  }
+
+  errorCallback(error) {
```

Comment:
rename to handleLocationError(error)

Reply:
https://github.com/VasiliiDubik/Vasiliidubik-weatherForecast/pull/1/commits/3e5fbf17f5f67085000f27605786ee9a3ba3477d